### PR TITLE
stats for redirected templates and deprecated properties

### PR DESCRIPTION
**Pull request by Nono314**
- fixed existing code for highlighting mapped templates that have been removed or have redundant mappings between redirected templates: bug [here] (https://github.com/Nono314/extraction-framework/commit/5c872f0ed3d0841c876790c4268c8c5a93d14b2a#diff-e373dc402eec9969c2d9027bca28464dL54) (the intersection was always empty since redirects hold templates with namespace while mappings has them without).
- added a new column to display for each template the number of mapped properties that are not found

### CODE REVIEW

File: server/src/main/scala/org/dbpedia/extraction/server/stats/MappingStats.scala [hunk 1]
```diff
@@ -24,23 +24,28 @@ class MappingStats(templateStats: TemplateStats, val templateName: String, val i
    * @param all count all properties or only the ones that are mapped
    * @param use count property use in wikipedia articles or just the definition in template
```
Comment (jimkont) [inline server/src/main/scala/org/dbpedia/extraction/server/stats/MappingStats.scala]: Can you add the new parameter documentation here? and probably make invalid false by default?

File: server/src/main/scala/org/dbpedia/extraction/server/stats/MappingStats.scala [hunk 2]
```diff
@@ -24,23 +24,28 @@ class MappingStats(templateStats: TemplateStats, val templateName: String, val i
...
-  val propertyCount = count(true, false)
-  val mappedPropertyCount = count(false, false) 
+  val propertyCount = count(true, false, false)
+  val mappedPropertyCount = count(false, false, false) 
 
-  val propertyUseCount = count(true, true)
-  val mappedPropertyUseCount = count(false, true) 
+  val propertyUseCount = count(true, true, false)
+  val mappedPropertyUseCount = count(false, true, false) 
+  val mappedPropertyUnuseCount = count(false, false, true)
```
Comment (jimkont) [inline server/src/main/scala/org/dbpedia/extraction/server/stats/MappingStats.scala]: maybe skip the invalid argument is false is default in the previous initializations and here declare with `count(false, false, invalid = true)` 
it makes it easier to read

File: server/src/main/scala/org/dbpedia/extraction/server/stats/MappingStatsHolder.scala [hunk 3]
```diff
@@ -51,7 +51,7 @@ object MappingStatsHolder {
         }
       }
       
-      val redirects = wikiStats.redirects.filterKeys(title => templateMappings.contains(title)).map(_.swap)
+      val redirects = wikiStats.redirects.filterKeys(title => templateMappings.contains(title.substring(templateNamespace.length))).map(_.swap)
```
Comment (jimkont) [inline server/src/main/scala/org/dbpedia/extraction/server/stats/MappingStatsHolder.scala]: is this related to this PR or a bug fix?

File: server/src/main/scala/org/dbpedia/extraction/server/resources/TemplateStatistics.scala [hunk 4]
```diff
@@ -170,7 +171,7 @@ class   TemplateStatistics(@PathParam("lang") langCode: String, @QueryParam("p")
             var redirectMsg = ""
             for (redirect <- targetRedirect) {
               if (mappingStat.isMapped) {
-                  //redirectMsg = " NOTE: the mapping for " + wikiDecode(redirect, language).substring(createMappingStats.templateNamespace.length()) + " is redundant!"
+                  redirectMsg = " NOTE: the mapping for " + redirect.substring(manager.templateNamespace.length) + " is redundant!"
```
Comment (jimkont) [inline server/src/main/scala/org/dbpedia/extraction/server/resources/TemplateStatistics.scala]: Just tested this locally and looks quite good! what would be nice is to make the title a link.
We also use bootstrap css and enclosing this into an alert or a warning would look better
http://getbootstrap.com/2.3.2/components.html#alerts

**Comment by jimkont** [comment]
Hi @Nono314 , very good! I made some minor comments inline.

File: server/src/main/scala/org/dbpedia/extraction/server/stats/MappingStatsHolder.scala (hunk 3 above)
Comment (Nono314) [inline server/src/main/scala/org/dbpedia/extraction/server/stats/MappingStatsHolder.scala]: actually both :) it's the part of the PR about fixing existing code.
I just realized I have explained the whereabouts in the issue about redirected templates but not in the PR itself... updating with that.

File: server/src/main/scala/org/dbpedia/extraction/server/resources/TemplateStatistics.scala (hunk 4 above)
Comment (Nono314) [inline server/src/main/scala/org/dbpedia/extraction/server/resources/TemplateStatistics.scala]: I also tested it locally and tried a link but links to redirected templates statistics just throw an exception. So it should be a link to the mapping page which is not really consistent with the column content... 
It was a quick fix to just make the current code work, I'll have a look at integrating it more smoothly in the page.

Comment (jimkont) [inline server/src/main/scala/org/dbpedia/extraction/server/resources/TemplateStatistics.scala]: I see, makes sense. Then maybe enclose it with a different css class to make it easier for someone to copy the title

File: server/src/main/scala/org/dbpedia/extraction/server/stats/MappingStats.scala (hunk 2 above)
Comment (Nono314) [inline server/src/main/scala/org/dbpedia/extraction/server/stats/MappingStats.scala]: Sure this was quite quick and dirty, it will be much cleaner this way

File: server/src/main/scala/org/dbpedia/extraction/server/resources/TemplateStatistics.scala (hunk 4 above)
Comment (Nono314) [inline server/src/main/scala/org/dbpedia/extraction/server/resources/TemplateStatistics.scala]: I have tried alerts but am not convinced. They use the same color codes already used, so they're either not standing out or confusing. They also add some unwanted padding. Also tried labels but they are a bit too flashy compared to the current tone.

**Comment by Nono314** [comment]
One thing that would be nice is having something more elaborate than just swapping the key/value pairs for redirects. This obviously only keeps one redirected template and you have to fix one to discover the next.
